app/views/posts/index.html.erb

```diff
@@ -0,0 +1,49 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Posts</h1>
+<% if user_signed_in? %>
+<p>
+  <%= current_user.name %> now you can see the secrets!
+</p>
+<p>
+  <%= link_to "Logout", destroy_user_session_path, method: :delete %>
+</p>
+<% else %>
+<p>
+  <%= link_to 'Sign Up', new_user_registration_path %> or <%= link_to 'Sign In', new_user_session_path %> to see the secrets
+</p>
+<% end %>
+
+<table>
+  <thead>
+    <tr>
+      <th>Title</th>
+      <th>Body</th>
+      <th>Author</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+    <tr>
+      <td><%= post.title %></td>
+      <td><%= post.body %></td>
+      
+      <% if user_signed_in? %>
+      <td><%= post.user.name %></td>
+      <!-- <td><%#= link_to 'Edit', edit_post_path(post) %></td> -->
+        <% if current_user.name == post.user.name %>
+          <td><%= link_to 'Delete', post, method: :delete %></td>
+        <% end %>
+      <% else %>
+      <td>Anonymous</td>
+      <% end %>
```

Comment:
These are logic that should be encapsulated in a helper. Check my comments for details.